README.md

```diff
@@ -1,2 +1,407 @@
+![badge](https://github.com/BlinkPay/Blink-Debit-API-Client-Java/actions/workflows/workflow.yml/badge.svg)
+
+This SDK allows merchants with Java-based e-commerce site to integrate with Blink PayNow and Blink AutoPay.
+
+# Minimum Requirements
+- Maven 3 or Gradle 7
+- Java 8
+- Lombok 1.18
+
+This SDK internally uses WebClient, a reactive web client introduced in Spring Framework 5, for making API calls.
+
+# Dependency
+## Maven
+```xml
+<dependency>
+    <groupId>nz.co.blinkpay</groupId>
+    <artifactId>blink-debit-api-client-java</artifactId>
+    <version>1.0.0</version>
+</dependency>
+```
+## Gradle
+```
+implementation 'nz.co.blinkpay:blink-debit-api-client-java:1.0.0
+```
+
+# Configuration
+- Customise/supply the required properties in your `application.yaml` or `application.properties`. This file should be available in your classpath, i.e. normally placed in `src/main/resources`.
+- Sandbox debit URL is `https://sandbox.debit.blinkpay.co.nz` and production debit URL is `https://debit.blinkpay.co.nz`.
+- The client credentials will be provided to you as part of the on-boarding process. These properties can also be supplied using environment variables.
+> **Warning** Take care not to check in your client ID and secret to your source control.
+```yaml
+blinkpay:
+  debit:
+    url: ${BLINKPAY_DEBIT_URL}
+  max:
+    connections: ${BLINKPAY_MAX_CONNECTIONS:10}
+    idle:
+      time: ${BLINKPAY_MAX_IDLE_TIME:PT20S}
+    life:
+      time: ${BLINKPAY_MAX_LIFE_TIME:PT60S}
+  pending:
+    acquire:
+      timeout: ${BLINKPAY_PENDING_ACQUIRE_TIMEOUT:PT10S}
+  eviction:
+    interval: ${BLINKPAY_EVICTION_INTERVAL:PT60S}
+  client:
+    id: ${BLINKPAY_CLIENT_ID}
+    secret: ${BLINKPAY_CLIENT_SECRET}
+```
+```properties
+blinkpay.debit.url=${BLINKPAY_DEBIT_URL}
+blinkpay.max.connections=${BLINKPAY_MAX_CONNECTIONS:10}
+blinkpay.max.idle.time=${BLINKPAY_MAX_IDLE_TIME:PT20S}
+blinkpay.max.life.time=${BLINKPAY_MAX_LIFE_TIME:PT60S}
+blinkpay.pending.acquire.timeout=${BLINKPAY_PENDING_ACQUIRE_TIMEOUT:PT10S}
+blinkpay.eviction.interval=${BLINKPAY_EVICTION_INTERVAL:PT60S}
+blinkpay.client.id=${BLINKPAY_CLIENT_ID}
+blinkpay.client.secret=${BLINKPAY_CLIENT_SECRET}
+# for non-Spring consumer as an alternative to spring.profiles.active property
+blinkpay.active.profile=${BLINKPAY_ACTIVE_PROFILE}
+```
+
+# Integration
+## Client
+Spring-based client code can simply autowire/inject the API client.
+```java
+@Autowired
+BlinkDebitClient client;
+```
+Pure Java client code can load the contents of `application.properties` into Properties.
+```java
+Properties properties = new Properties();
+properties.load(getClass().getClassLoader().getResourceAsStream("application.properties"));
+
+BlinkDebitClient client = new BlinkDebitClient(properties);
+```
+Optional correlation ID can be added as the last argument to API calls.
+
+## Bank Metadata
+```java
+List<BankMetadata> bankMetadataList = client.getMeta();
+```
+
+## Single/One-Off Consents
```

Comment:
Let's include a note:
"Except for quick-payments, the completion of a payment requires a consent in authorised status, followed by a payment"